src/cargo/core/resolver/resolve.rs

```diff
@@ -97,6 +98,21 @@ impl ResolveVersion {
     pub fn max_stable() -> ResolveVersion {
         ResolveVersion::V3
     }
+
+    /// Gets the default lockfile version for the given Rust version.
+    pub fn with_rust_version(rust_version: Option<&RustVersion>) -> Self {
```

Comment:
Should we put this behind `-Zmsrv-policy` or insta-stable?

I'm fine with insta-stable.  The main policy questions would be around
- Whether this should respect `--ignore-rust-version`, where present
- A workspace has no `rust-version`, so this is determined by finding the minimum `package.rust-version` among all workspace members

Reply:
Don't feel like it's a thing `-Zmsrv-policy` wants to address. If we had got MSRV-aware resolver when we started versioning lockfile, I believe we would have added this.

Regarding `--ignore-rust-version`, I am pretty sold on respecting that, given the semantic is like "hey I want Cargo to work as if I never set `rust-version`".

Reply:
I'm fine with that.

Reply:
After looking into the current codebase, I am now more unsure about supporting `--ignore-rust-version` for generating lockfile. If we aim to do that, we may need to expose `--ignore-rust-version` to every command that generates lockfiles.

* `cargo build` and other commands involving build
* `cargo fix`
* `cargo generate-lockfile`
* `cargo tree`
* `cargo metadata`
* `cargo update`

It's a cognitive overload for users to learn they have the control over lockfile versions for all these commands. The user experience would be better if we just provide it out-of-the-box. Lockfile versions are somehow opaque to users to me.

I cannot think of any situation people may opt-in `--ignore-rust-version` for controlling the lockfile version. It's incompatible when you want to support old versions but requesting for a new lockfile version. For people want old lockfiles, unless there are bugs in newer versions, otherwise we don't want people to continue using old versions. And they are alwasy able to set `rust-version` for older lockfiles.

Moreover, the workaround is always there — manually changing the `version` field in the lockfile.

Is there any use case I am not aware of for respecting `--ignore-rust-version` for lockfile generating?

Reply:
We could start respecting `rust-version` from v4, so `rust-version` older than `1.75` won't affect lockfile versions. This would be less controversial.

Reply:
This is reminding me of an idea I had for MSRV-aware resolver of putting the MSRV in the lockfile and making it sticky.  Having to pass flags along for every call can be pretty frustrating.